test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala

```diff
@@ -16,35 +16,60 @@
 
 package uk.gov.hmrc.apinotificationqueue.controller
 
+import org.scalatest.mockito.MockitoSugar
 import play.api.http.Status
+import play.api.mvc.{AnyContentAsEmpty, Headers, Result}
 import play.api.test.FakeRequest
+import uk.gov.hmrc.apinotificationqueue.service.QueueService
+import uk.gov.hmrc.http.BadRequestException
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
-class QueueControllerSpec extends UnitSpec with WithFakeApplication {
+import scala.concurrent.Future
 
+class QueueControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplication {
+
+  trait Setup {
+    val mockQueueService = mock[QueueService]
+    val underTest = new QueueController(mockQueueService)
+  }
+
+
+  "POST /queue" should {
+    "throw exception if x-client-id not present" in new Setup {
+      intercept[BadRequestException] {
+        underTest.save()(FakeRequest("POST", "/queue"))
+      }
+    }
+    "throw exception if no payload" in new Setup {
+      private val request = FakeRequest("POST", "/queue", Headers("x-client-id" -> "a"), AnyContentAsEmpty)
+      intercept[BadRequestException] {
+        val result: Future[Result] = underTest.save()(request)
+      }
+    }
+    "return 201 if body and headers" in new Setup {
```

Comment:
I think `return 201 if body and headers present` rather than `return 201 if body and headers`